web/app/scripts/src/directives/form.es.js

```diff
@@ -39,11 +50,49 @@ module.directive('surveysForm', [
                             // been set.
                             if (Object.keys(scope.model).length === 0) {
                                 scope.model = FieldsetParser.parseFormToModel(structure);
+                                countQuestionsTotal();
                             }
                         });
                     }
                 });
 
+                // answers is an object containing objects
+                // object {
+                //     1 : {
+                //         2 : 0
+                //         3 : 0
+                //     }
+                // }
+                // answered is a number of type number
+                const countNumberOfAnsweredQuestions = function (answers) {
+                    let answered = 0;
+
+                    for (let groupKey in answers) {
+                        let answerGroup = answers[groupKey]
+                        for (let answerKey in answerGroup) {
+                            let answer = answerGroup[answerKey]
+                            if (answer > 0 || answer.length > 0) {
+                                answered++;
+                            }
+                        }
+                    }
+
+                    return answered;
+                };
+
+                const calculatePercentageComplete = function (completedQuestions) {
+                    const result = ((completedQuestions / totalQuestionCount) * 100)
+                    if (!isNaN(result)) {
+                        scope.percentageComplete = Math.round(result) + '%';
+                    }
```

Comment:
To not make it flash it should be hidden with `ng-show`. I think if there aren't any questions then the result should be `100%` because you can't answer anything? If ti would be `0%` that would mean the user needs to answer something but nothing is there so that would be confusing imo. But maybe ask someone else